code/game/objects/items/shooting_range.dm

```diff
@@ -77,101 +76,36 @@
 	desc = "A shooting target that looks like a xenomorphic alien."
 	hp = 2350 // alium onest too kinda
 
-/obj/item/target/bullet_act(var/obj/item/projectile/Proj)
-	var/p_x = Proj.p_x + pick(0,0,0,0,0,-1,1) // really ugly way of coding "sometimes offset Proj.p_x!"
-	var/p_y = Proj.p_y + pick(0,0,0,0,0,-1,1)
-	var/decaltype = 1 // 1 - scorch, 2 - bullet
-
-	if(istype(/obj/item/projectile/bullet, Proj))
-		decaltype = 2
-
-
-	virtualIcon = new(icon, icon_state)
-
-	if( virtualIcon.GetPixel(p_x, p_y) ) // if the located pixel isn't blank (null)
-
-		hp -= Proj.damage
+#define DECALTYPE_SCORCH 1
+#define DECALTYPE_BULLET 2
+
+/obj/item/target/bullet_act(obj/item/projectile/P)
+	var/p_x = P.p_x + pick(0,0,0,0,0,-1,1) // really ugly way of coding "sometimes offset P.p_x!"
+	var/p_y = P.p_y + pick(0,0,0,0,0,-1,1)
+	var/decaltype = DECALTYPE_SCORCH
+	if(istype(P, /obj/item/projectile/bullet))
+		decaltype = DECALTYPE_BULLET
+	var/icon/C = icon(icon, icon_state)
+	if(C.GetPixel(p_x, p_y) && LAZYLEN(overlays) <= 35) // if the located pixel isn't blank (null)
+		hp -= P.damage
 		if(hp <= 0)
-			visible_message("<span class='warning'>[src] breaks into tiny pieces and collapses!</span>")
+			visible_message("<span class='danger'>[src] breaks into tiny pieces and collapses!</span>")
 			qdel(src)
-
-		// Create a temporary object to represent the damage
-		var/obj/bmark = new
-		bmark.pixel_x = p_x
-		bmark.pixel_y = p_y
-		bmark.icon = 'icons/effects/effects.dmi'
-		bmark.layer = 3.5
-		bmark.icon_state = "scorch"
-
-		if(decaltype == 1)
-			// Energy weapons are hot. they scorch!
-
-			// offset correction
-			bmark.pixel_x--
-			bmark.pixel_y--
-
-			if(Proj.damage >= 20 || istype(Proj, /obj/item/projectile/beam/practice))
-				bmark.icon_state = "scorch"
-				bmark.dir = pick(NORTH,SOUTH,EAST,WEST) // random scorch design
-
-
+			return
+		var/image/bullet_hole = image('icons/effects/effects.dmi', "scorch", OBJ_LAYER + 0.5)
+		bullet_hole.pixel_x = p_x - 1 //offset correction
+		bullet_hole.pixel_y = p_y - 1
+		if(decaltype == DECALTYPE_SCORCH)
+			if(P.damage >= 20 || istype(P, /obj/item/projectile/beam/practice))
+				bullet_hole.setDir(pick(NORTH,SOUTH,EAST,WEST))// random scorch design. light_scorch does not have different directions
```

Comment:
This one was duped in TG before line 99. Issue is that only scorch has 4 directions and light_scorch does not. TG has the same icon.